base/regex.jl

```diff
@@ -285,6 +285,11 @@ iterate(m::RegexMatch, args...) = iterate(m.captures, args...)
 length(m::RegexMatch) = length(m.captures)
 eltype(m::RegexMatch) = eltype(m.captures)
 
+NamedTuple(m::RegexMatch) = NamedTuple{Symbol.(Tuple(keys(m)))}(values(m))
```

Comment:
This seems guaranteed to be rather slow, so I am not sure it is generally very useful
```suggestion
```

Reply:
Operating with NamedTuples is much slower than Dict if the type are known to the compiler, and this is case where the types cannot be known to the compiler

Reply:
Only calling btime would be fast in that case because you are only ever matching the regex once. That is not realistic for using a regex. A more accurate measurement might be at least putting them all together
```
for DT in Any[values, Dict, NamedTuple, Tuple];
  @benchmark sum(x->parse(Int, x), $DT(match(r"(?<a>.)(?<c>.)(?<b>.)", "123")))
end
```